providers/ms365/resources/tenant.go

```diff
@@ -256,3 +284,129 @@ func (a *mqlMicrosoftTenant) formsSettings() (*mqlMicrosoftTenantFormsSettings,
 
 	return formSetting.(*mqlMicrosoftTenantFormsSettings), nil
 }
+
+func (a *mqlMicrosoftTenant) fetchRealmData() error {
+	a.lock.Lock()
+	defer a.lock.Unlock()
+
+	if a.realmDataFetched {
+		return a.realmDataErr
+	}
+	a.realmDataFetched = true
+
+	verifiedDomains := a.GetVerifiedDomains()
+	if verifiedDomains.Error != nil {
+		a.realmDataErr = verifiedDomains.Error
+		return a.realmDataErr
+	}
+
+	if len(verifiedDomains.Data) == 0 {
+		a.realmInfos = []any{}
+		a.brandingInfos = []any{}
+		return nil
+	}
+
+	var realmInfos []any
+	var brandingInfos []any
+
+	for _, domain := range verifiedDomains.Data {
+		domainDict, ok := domain.(map[string]any)
+		if !ok {
+			continue
+		}
+
+		domainName, ok := domainDict["name"].(string)
+		if !ok || domainName == "" {
+			continue
+		}
+
+		realmResp, err := a.getUserRealmInfoForDomain(domainName)
+		if err != nil {
+			logger.DebugDumpJSON("tenant-realm-domain-error", []byte(fmt.Sprintf("domain: %s, error: %v", domainName, err)))
+			a.realmDataErr = err
+			return err
+		}
+
+		// Create realm info resource
+		realmId := fmt.Sprintf("%s-realm-info-%s", a.Id.Data, realmResp.DomainName)
+		mqlRealm, err := CreateResource(a.MqlRuntime, ResourceMicrosoftTenantRealmInfo,
+			map[string]*llx.RawData{
+				"__id":                llx.StringData(realmId),
+				"federationBrandName": llx.StringData(realmResp.FederationBrandName),
+				"domainName":          llx.StringData(realmResp.DomainName),
+				"login":               llx.StringData(realmResp.Login),
+				"nameSpaceType":       llx.StringData(realmResp.NameSpaceType),
+			})
+		if err != nil {
+			logger.DebugDumpJSON("tenant-realm-create-error", []byte(fmt.Sprintf("domain: %s, error: %v", domainName, err)))
+			a.realmDataErr = err
+			return err
+		}
+		realmInfos = append(realmInfos, mqlRealm)
+
+		// Create branding info resources
+		for i, info := range realmResp.TenantBrandingInfo {
+			brandingId := fmt.Sprintf("%s-branding-info-%s-%d", a.Id.Data, realmResp.DomainName, i)
+			mqlBranding, err := CreateResource(a.MqlRuntime, ResourceMicrosoftTenantBrandingInfo,
+				map[string]*llx.RawData{
+					"__id":                   llx.StringData(brandingId),
+					"domainName":             llx.StringData(realmResp.DomainName),
+					"keepMeSignedInDisabled": llx.BoolData(info.KeepMeSignedInDisabled),
+					"backgroundColor":        llx.StringData(info.BackgroundColor),
+					"bannerLogo":             llx.StringData(info.BannerLogo),
+					"useTransparentLightbox": llx.BoolData(info.UseTransparentLightbox),
+				})
+			if err != nil {
+				logger.DebugDumpJSON("tenant-branding-create-error", []byte(fmt.Sprintf("domain: %s, index: %d, error: %v", domainName, i, err)))
+				a.realmDataErr = err
+				return err
+			}
+			brandingInfos = append(brandingInfos, mqlBranding)
+		}
+	}
+
+	a.realmInfos = realmInfos
```

Comment:
I think we should set the state with the resources here, see https://github.com/mondoohq/cnquery/blob/3b1546508fd2501046c2ca70d6e31c8ab56bd942/providers/os/resources/sshd.go#L202-L230 as an example